_posts/2022/2022-01-28-working-with-shaders.md

```diff
@@ -0,0 +1,229 @@
+
+<iframe src = "https://www.shadertoy.com/embed/fdBcWz" width="100%" height="360" ></iframe>
+
+
+
+One can also use [Manhattan distance](https://en.wikipedia.org/wiki/Manhattan_distance) as our distance function [like here](https://www.shadertoy.com/view/NdScDR). I just used some good old matrix transformation for rotating it.
+**Result:**
+<iframe src = "https://www.shadertoy.com/embed/NdScDR" width="100%" height="360"></iframe>
+
+  
+
+We can also use [smooth steps](https://thebookofshaders.com/glossary/?search=smoothstep).
+
+  
+
+## Water Shader
+
+A straightforward water shader is just using distortion. We can use a du/dv map and move it to create distortion. We can then add two or three maps to our uv values and mixed reflection and refraction textures to give a [satisfying result](https://www.shadertoy.com/view/7dcGzf).
+<iframe src = "https://www.shadertoy.com/embed/7dcGzf" width="100%" height="360"></iframe>
+
+  
+
+Let’s try to make something more cartoonish.
+
+  
+
+Meet Voronoi Diagrams.
+
+  
+
+Voronoi diagrams are excellent. Essentially color of a specific point in a plane depends on its nearest point. This coloring divides the plane into multiple segments.
+
+
+<img src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/54/Euclidean_Voronoi_diagram.svg/1920px-Euclidean_Voronoi_diagram.svg.png" alt="drawing" width="500"/>
+
+We can also generalise it to any metric possible, giving aesthetically pleasing results. You only need a set of points and a metric, and for that, there would exist a unique Voronoi diagram.
+
+  
+
+They are cool [do check them out](https://en.wikipedia.org/wiki/Voronoi_diagram)
+
+  
+
+So let’s make a Voronoi diagram and have the points move! Color it, and there is your cartoonish [water shader](https://www.shadertoy.com/view/NdVXRG).
+
+<iframe src = "https://www.shadertoy.com/embed/NdVXRG" width="100%" height="360"></iframe>
+  
+
+## Fire Shader
+
+I decided to use Fractal Brownian Motion on simplex noise for this. Then I added a gradient and have `uv.y` continuously decreasing, giving the illusion that the fire is moving upwards. Yeah, it’s certainly complicated, so let’s go over them one by one. 
+
+  
+
+**Simplex noise** is a type of noise, like Perlin noise. The function noise generates this.
+
+  
+
+**The Fractal Brownian Motion** combines multiple steps of Simplex Noise (or another similar noise function), each with a different frequency and amplitude.
+
+ [Here is the result](https://www.shadertoy.com/view/7tc3zs) left is simplex noise and right is fractal brownian motion.
+<iframe src = "https://www.shadertoy.com/embed/NtGGRc" width="100%" height="360"></iframe>
+
+We color this and [we're done](https://www.shadertoy.com/view/7tc3zs).
+
+
+  
+
+  
+
+## Cel shader
+
+Playing Legend of Zelda is an experience. The graphics are aesthetically beautiful. It just feels different, but why is that? One can tell it is not going for realistic looks but rather making it cartoonish.
```

Comment:
graphics that don't aim to be realistic but try to achieve a _style_ are referred to as `stylized graphics`